src/components/cl/doca_urom/cl_doca_urom.h

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2020, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCC_CL_DOCA_UROM_H_
+#define UCC_CL_DOCA_UROM_H_
+
+#include "components/cl/ucc_cl.h"
+#include "components/cl/ucc_cl_log.h"
+#include "components/tl/ucc_tl.h"
+#include "coll_score/ucc_coll_score.h"
+#include "utils/ucc_mpool.h"
+
+#include <doca_dev.h>
+#include <doca_urom.h>
+#include <doca_pe.h>
+#include <doca_ctx.h>
+#include <doca_buf.h>
+
+#include <worker_ucc.h>
+#include <common.h>
+#include <urom_common.h>
+#include <urom_ucc.h>
+
+#ifndef UCC_CL_DOCA_UROM_DEFAULT_SCORE
+#define UCC_CL_DOCA_UROM_DEFAULT_SCORE 100
+#endif
+
+#define UCC_CL_DOCA_UROM_ADDR_MAX_LEN  1024
+#define UCC_CL_DOCA_UROM_MAX_TEAMS     16
+
+typedef struct ucc_cl_doca_urom_iface {
+    ucc_cl_iface_t super;
+} ucc_cl_doca_urom_iface_t;
+
+// Extern iface should follow the pattern: ucc_cl_<cl_name>
+extern ucc_cl_doca_urom_iface_t ucc_cl_doca_urom;
+
+typedef struct ucc_cl_doca_urom_lib_config {
+    ucc_cl_lib_config_t super;
+    char *device;
+    char *plugin_name;
+    int doca_log_level;
+} ucc_cl_doca_urom_lib_config_t;
+
+typedef struct ucc_cl_doca_urom_context_config {
+    ucc_cl_context_config_t super;
+    ucs_config_names_array_t plugin_envs;
```

Comment:
is there a reason to have `plugin_envs` and `plugin_name` at different hierarchical level (lib and ctx)?